### DISCUSSION TITLE
Preserve data with client useState on server/api

### DISCUSSION THREAD

**Discussion by procrates** [nuxt/nuxt#18039] [Questions] [answered]
Hello! Been using v2 for quite some time and I was really excited to start using v3! And I must say it is AWESOME!

This is the first question I have to ask becouse I can't find any docs on it or Googles...

I have a form and I am using useState to make it reactive as it is recommended over ref. 

I see that I can use those useState's on the server/api. What is the best practice on doing so? 

Now I am using $fetch in a submit function and sending the states with body:{} and getting the body on the server using useBody and I can actually see the states, but I also see all the other states I am using.. How is that? 

I will share an example when I have access to the code. 

But it looks something like this:

::Edit::

On client:
```
const name = useState("name", () => "name")
const type = useState("type", () => "type")
const body = useState("body", () => "body")
const createPost = async () => {
    const data = await $fetch("/api/create/blog", {
        method: "POST",
        body: {
            name: name,
            type: type,
            body: body
        }
    })
}
```
On server/api result of `useBody`:
```
/* Get body data on server */
  const body = await useBody(req);
  console.log("useBody", body);

/** Terminal **/

useBody {
  name: {
    _object: {
      mobileNavState: false,
      email: 'test@test.com',
      password: 'test123',
      name: 'name',
      type: 'type',
      body: 'body'
    },
    _key: 'name',
    __v_isRef: true
  },
  type: {
    _object: {
      mobileNavState: false,
      email: 'test@test.com',
      password: 'test123',
      name: 'name',
      type: 'type',
      body: 'body'
    },
    _key: 'type',
    __v_isRef: true
  },
  body: {
    _object: {
      mobileNavState: false,
      email: 'test@test.com',
      password: 'test123',
      name: 'name',
      type: 'type',
      body: 'body'
    },
    _key: 'body',
    __v_isRef: true
  }
}
```
As you can see when sending useState it sends all the states in every instance of use state. I dont know if this is intended, but how would i be able to do this is the best way. This is a useful feature for me so it would be nice to have an easy way to destructure it to use the values.

::End edit::
It looks like all the useState's is attached to all the props I am sending to server. 

Can someone please explain? And please give an example on how to send just the form data:) 

Thank you very much!

**Comment by wobsoriano** [comment, marked as answer]
As far as I know, you can't use `useState` on `server/api`. 

You can check a sample implementation of server/client state [here](https://stackblitz.com/github/nuxt/framework/tree/main/examples/locale?terminal=dev).

**Reply by procrates** [reply]
Oh man... I forgot to use .value to get the value xD...

But it's still nice to see what happend when sending a useState function to the server.
